packages/language/src/preprocessor/instruction-interpreter.ts

```diff
@@ -473,13 +479,82 @@ function runInstructionSync(
     case inst.InstructionKind.Deactivate:
       runDeactivateInstruction(instruction, context);
       break;
+    case inst.InstructionKind.Note:
+      runNoteInstruction(instruction, context);
+      break;
     case inst.InstructionKind.Halt:
       runHaltInstruction(instruction, context);
       break;
   }
   return undefined;
 }
 
+function runNoteInstruction(
+  instruction: inst.NoteInstruction,
+  context: InterpreterContext,
+): void {
+  const message = valueToString(
+    evaluateExpression(instruction.message, context),
+  );
+  const codeValue = evaluateExpression(instruction.code, context);
+  const code = valueToNumber(codeValue);
+  let error: Diagnostic = undefined!;
```

Comment:
I don' think this is necessary.

```suggestion
  let error: Diagnostic;
```